examples/cpp_execution/text_inference_using_cpp.py

```diff
@@ -0,0 +1,251 @@
+#
+# -----------------------------------------------------------------------------
+import argparse
+import logging
+import os
+import sys
+from pathlib import Path
+from typing import List, Optional, Union
+
+from transformers import PreTrainedTokenizer, PreTrainedTokenizerFast
+
+import QEfficient
+from QEfficient.cloud.export import get_onnx_model_path
+from QEfficient.generation.text_generation_inference import fix_prompts, get_compilation_dims, get_input_prompts
+from QEfficient.utils import check_and_assign_cache_dir, get_qpc_dir_path, load_hf_tokenizer, qpc_exists
+from QEfficient.utils.logging_utils import logger
+
+script_dir = Path(__file__).resolve().parent
+so_folder_path = script_dir / "build"
+
+if so_folder_path.is_dir():
+    sys.path.append(str(so_folder_path))
+    try:
+        import InferenceSetIOBuffer  # noqa: E402
+    except ImportError:
+        logger.error("Error importing InferenceSetIOBuffer Module")
+else:
+    raise FileNotFoundError("Please follow README instructions to first compile the cpp files")
+
+
+def main(
+    model_name: str,
+    num_cores: int,
+    device_group: Optional[List[int]] = None,
+    prompt: Optional[str] = None,  # type: ignore
+    prompts_txt_file_path: Optional[str] = None,
+    aic_enable_depth_first: bool = False,
+    mos: int = -1,
+    batch_size: int = 1,
+    full_batch_size: Optional[int] = None,
+    prompt_len: int = 32,
+    ctx_len: int = 128,
+    generation_len: Optional[int] = None,
+    mxfp6: bool = False,
+    mxint8: bool = False,
+    local_model_dir: Optional[str] = None,
+    cache_dir: Optional[str] = None,
+    hf_token: Optional[str] = None,
+) -> None:
+    """
+    1. Check if compiled qpc for given config already exists, if it does jump to execute, else
+    2. Check if exported ONNX file already exists, if true, jump to compilation -> execution, else
+    3. Check if HF model exists in cache, if true, start transform -> export -> compilation -> execution, else,
+    4. Download HF model -> transform -> export -> compile -> execute
+    ``Mandatory`` Args:
+        :model_name (str): Hugging Face Model Card name, Example: ``gpt2``
+        :num_cores (int): Number of cores to compile model on.
+    ``Optional`` Args:
+        :device_group (List[int]): Device Ids to be used for compilation. If ``len(device_group) > 1``, multiple Card setup is enabled. ``Defaults to None.``
+        :prompt (str): Sample prompt for the model text generation. ``Defaults to None.``
+        :prompts_txt_file_path (str): Path to txt file for multiple input prompts. ``Defaults to None.``
+        :aic_enable_depth_first (bool): Enables ``DFS`` with default memory size. ``Defaults to False.``
+        :mos (int): Effort level to reduce the on-chip memory. ``Defaults to -1.``
+        :batch_size (int): Batch size to compile the model for. ``Defaults to 1.``
+        :full_batch_size (int): Set full batch size to enable continuous batching mode. ``Default to None``
```

Comment:
While removing the full_batch_size args please remove it from the doc string as well.